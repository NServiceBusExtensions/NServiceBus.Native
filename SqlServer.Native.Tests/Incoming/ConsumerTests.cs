using System.Collections.Generic;
using NServiceBus.Transport.SqlServerNative;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class ConsumerTests : TestBase
{
    string table = "ConsumerTests";

    [Fact]
    public void Single_bytes()
    {
        TestDataBuilder.SendData(table);
        var consumer = new Consumer(table);
        var result = consumer.ConsumeBytes(Connection.ConnectionString).Result;
        ObjectApprover.VerifyWithJson(result);
    }

    [Fact]
    public void Single_bytes_nulls()
    {
        TestDataBuilder.SendNullData(table);
        var consumer = new Consumer(table);
        var result = consumer.ConsumeBytes(Connection.ConnectionString).Result;
        ObjectApprover.VerifyWithJson(result);
    }

    [Fact]
    public void Single_stream()
    {
        TestDataBuilder.SendData(table);
        var consumer = new Consumer(table);
        using (var result = consumer.ConsumeStream(Connection.ConnectionString).Result)
        {
            ObjectApprover.VerifyWithJson(result.ToVerifyTarget());
        }
    }

    [Fact]
    public void Single_stream_nulls()
    {
        TestDataBuilder.SendNullData(table);
        var consumer = new Consumer(table);
        using (var result = consumer.ConsumeStream(Connection.ConnectionString).Result)
        {
            ObjectApprover.VerifyWithJson(result.ToVerifyTarget());
        }
    }

    [Fact]
    public void Batch_bytes()
    {
        TestDataBuilder.SendMultipleData(table);

        var consumer = new Consumer(table);
        var messages = new List<IncomingBytesMessage>();
        var result = consumer.ConsumeBytes(
                connection: Connection.ConnectionString,
                size: 3,
                action: message => { messages.Add(message); })
            .Result;
        Assert.Equal(3, result.LastRowVersion);
        Assert.Equal(3, result.Count);
        ObjectApprover.VerifyWithJson(messages);
    }

    [Fact]
    public void Batch_stream()
    {
        TestDataBuilder.SendMultipleData(table);

        var consumer = new Consumer(table);
        var messages = new List<object>();
        var result = consumer.ConsumeStream(
                connection: Connection.ConnectionString,
                size: 3,
                action: message => { messages.Add(message.ToVerifyTarget()); })
            .Result;
        Assert.Equal(3, result.Count);
        Assert.Equal(3, result.LastRowVersion);
        ObjectApprover.VerifyWithJson(messages);
    }

    public ConsumerTests(ITestOutputHelper output) : base(output)
    {
        SqlHelpers.Drop(Connection.ConnectionString, table).Await();
        QueueCreator.Create(Connection.ConnectionString, table).Await();
    }
}