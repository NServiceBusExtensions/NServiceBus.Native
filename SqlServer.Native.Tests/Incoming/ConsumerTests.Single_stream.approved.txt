{
  "CorrelationId": "theCorrelationId",
  "Expires": "2000-01-01T01:01:01",
  "Headers": "headers",
  "Id": "00000000-0000-0000-0000-000000000001",
  "ReplyToAddress": "theReplyToAddress",
  "RowVersion": 1,
  "bodyString": "{}"
}